### PR TITLE
[systemtest][fixes] Some fixes and add method for listing deployed resources

### PR DESCRIPTION
**Pull request by im-konge**
Signed-off-by: Lukas Kral <lukywill16@gmail.com>

### Type of change

- Bugfix

### Description

In nightlies we can see that in `KafkaST` if the `testDeployKafkaClusterViaTemplate` fails the template resources are not properly deleted, so I added the deletion for whole template and all pods with `openshift` name. The test failure and the skip of deletion caused that whole `KafkaST` was failing, so it ran again. This will (hopefully) fix this problem.

Second problem was with KafkaConnect topic deletion, where the `CONNECT_TOPIC` was bind to the connector created in the test, so the right after the `.withPropagationPolicy(DeletionPropagation.FOREGROUND).delete()` was executed, topic was recreated. So I changed the order of resource creation and removed the unnecessary deletions and waits, as this is done by `ResourceManager`

### Checklist

- [ ] Make sure all tests pass

### CODE REVIEW

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=KafkaST,ConnectST

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java [hunk 1]
```diff
@@ -106,40 +106,39 @@
 class KafkaST extends AbstractST {
 
     private static final Logger LOGGER = LogManager.getLogger(KafkaST.class);
-
+    private static final String TEMPLATE_PATH = TestUtils.USER_PATH + "/../examples/templates/cluster-operator";
     public static final String NAMESPACE = "kafka-cluster-test";
+    private static final String OPENSHIFT_CLUSTER_NAME = "openshift-my-cluster";
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:111]: why it's global constant if it's used in one test?

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java [hunk 2]
```diff
@@ -106,40 +106,39 @@
 class KafkaST extends AbstractST {
 
     private static final Logger LOGGER = LogManager.getLogger(KafkaST.class);
-
+    private static final String TEMPLATE_PATH = TestUtils.USER_PATH + "/../examples/templates/cluster-operator";
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:109]: why it's global constant if it's used in one test?

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java [hunk 3]
```diff
@@ -1957,6 +1956,14 @@ void setup() throws Exception {
     @Override
     protected void tearDownEnvironmentAfterEach() throws Exception {
         super.tearDownEnvironmentAfterEach();
+        if (cluster.getListOfDeployedResources().contains(TEMPLATE_PATH)) {
+            cluster.deleteCustomResources(TEMPLATE_PATH);
+        }
+
+        kubeClient().listPods().stream()
+            .filter(p -> p.getMetadata().getName().startsWith(OPENSHIFT_CLUSTER_NAME))
+            .forEach(p -> PodUtils.deletePodWithWait(p.getMetadata().getName()));
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:1965]: Why is this moved from the one scenario where everything is deployed (and deleted) to afterEach?

Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:1965]: If the test fails for example in Kafka creation (deploying), the resources are not deleted properly. This will ensure that all resources from the `testDeployKafkaClusterViaTemplate` will be removed and will not affect other tests.

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java (hunk 1 above)
Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:111]: I used this for the deletion, if you don't agree, I can remove it.

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java (hunk 2 above)
Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:109]: Same as above

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java (hunk 3 above)
Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:1965]: I was looking for some solution to be executed only for the `testDeployKafkaClusterViaTemplate()`, but there is no annotation for that. Only thing what I can add is the `try{}catch{}` block and that's not good approach from MPOV

Comment (Frawless) [reply systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:1965]: Oh make sense to me!

**Review by Frawless — APPROVED** [review]

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java (hunk 3 above)
Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:1965]: Maybe we can talk to someone from JUnit to add anno to do something after specific test :smile:

**Review by scholzj — APPROVED** [review]

**Review by ppatierno — APPROVED** [review]

**Comment by im-konge** [comment]
Need to do some investigation on new kind of cascading -> it seems that topic is recreated after the deletion.

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java (hunk 3 above)
Comment (michalxo) [reply systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:1965]: @After if testName = mySpecific test 
https://stackoverflow.com/a/41321322 ;-)

**Comment by im-konge** [comment]
@ppatierno @scholz do you want to do review again? I found out where the problem with `.withPropagationPolicy(DeletionPropagation.FOREGROUND).delete()` was -> the `CONNECT_TOPIC` was bind with the connector, so right after deletion the topic was re-created. 

So I changed the order of resource creation and remove unnecessary deletes and waits -> this is done by the ResourceManager.

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=KafkaST,ConnectST

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=KafkaST,ConnectST

**Review by scholzj — APPROVED** [review]

**Review by michalxo — APPROVED** [review]

**Comment by im-konge** [comment]
@strimzi-ci run tests true profile=regression testcase=io.strimzi.systemtest.kafka.KafkaST#testKafkaOffsetsReplicationFactorHigherThanReplicas,io.strimzi.systemtest.kafka.KafkaST#testSimpleDynamicConfiguration,io.strimzi.systemtest.kafka.KafkaST#testDynamicConfigurationExternalTls
